### DISCUSSION TITLE
is there any faster way to resamle 1min data to daily data

### DISCUSSION THREAD

**Discussion by zha0yangchen** [kernc/backtesting.py#687] [General]
my stratege need 1min data and daily data, if i use 1min data resample to daily data, the backtesting will be very slow,  i wonder  is there any faster way to resamle 1min data to daily data?

**Comment by kernc** [comment]
Can you show us how you are currently resampling it? It's best to resample it outside the Backtest altogether, and there are probably few faster ways than pandas' `df.resample('1D')` ...

**Reply by zha0yangchen** [reply]
hi kernc, thank you for your reply! now, i resampling daily bar from 1min bar under next function, every 1min bar coming, i'll resampling it to daily bar, as my strategy need 10 days bar data. i also tried to load day bar to backtesting before the backtest begin, but i still facing same issue, i need to upsampling the day bar to 1min bar, then resample to day bar

**Reply by kernc** [reply]
Resampling in `next()` sounds absolutely a wrong way to do it!

**Reply by zha0yangchen** [reply]
now, I try to use resample_apply to resample the 1min data, but the result is not as my expected, the result is like this:
[  nan   nan   nan ... 9865. 9865. 9865.] , there are a lot of repeat data in the result, how can i get a result without repeat data? or if the repeat data can be replace by nan, then I can dropna in my strategy, thank you.
